docker/Dockerfile.xpu

```diff
@@ -1,187 +1,84 @@
 ARG ONEAPI_VERSION=2025.1.3-0
+ARG VLLM_VERSION=3cd36660f72f75b888c82a8feac93ea9f17c8e1e
```

Comment:
Are we sure we want something ahead of [v0.11.1rc1](https://github.com/vllm-project/vllm/releases/tag/v0.11.1rc1)? I think from a project perspective were still staying on `v0.11.0` release because we are not prepared to pickup torch `2.9.0`. That being said the XPU image is independent so you can definetly chose to base off a commit ahead of the rest of the project, just wanted to call this out.